tests/test_working_copy_sqlserver.py

```diff
@@ -0,0 +1,313 @@
+            # but we don't care exactly what it is.
+            try:
+                colour_id_line = diff[-6]
+            except KeyError:
+                colour_id_line = ""
+
+            assert diff[-46:] == [
+                "--- nz_waca_adjustments:meta:schema.json",
+                "+++ nz_waca_adjustments:meta:schema.json",
+                "  [",
+                "    {",
+                '      "id": "79d3c4ca-3abd-0a30-2045-45169357113c",',
+                '      "name": "id",',
+                '      "dataType": "integer",',
+                '      "primaryKeyIndex": 0,',
+                '      "size": 64',
+                "    },",
+                "    {",
+                '      "id": "c1d4dea1-c0ad-0255-7857-b5695e3ba2e9",',
+                '      "name": "geom",',
+                '      "dataType": "geometry",',
+                '      "geometryType": "MULTIPOLYGON",',
+                '      "geometryCRS": "EPSG:4167"',
+                "    },",
+                "    {",
+                '      "id": "d3d4b64b-d48e-4069-4bb5-dfa943d91e6b",',
+                '      "name": "date_adjusted",',
+                '      "dataType": "timestamp"',
+                "    },",
+                "-   {",
+                '-     "id": "dff34196-229d-f0b5-7fd4-b14ecf835b2c",',
+                '-     "name": "survey_reference",',
+                '-     "dataType": "text",',
+                '-     "length": 50',
+                "-   },",
+                "    {",
+                '      "id": "13dc4918-974e-978f-05ce-3b4321077c50",',
+                '      "name": "adjusted_nodes",',
+                '      "dataType": "integer",',
+                '      "size": 32',
+                "    },",
+                "+   {",
+                colour_id_line,
+                '+     "name": "colour",',
+                '+     "dataType": "text",',
+                '+     "length": 32',
+                "+   },",
+                "  ]",
+            ]
+
+            orig_head = repo.head.peel(pygit2.Commit).hex
+
+            r = cli_runner.invoke(["commit", "-m", "test_commit"])
+            assert r.exit_code == 0, r.stderr
+
+            r = cli_runner.invoke(["status"])
+            assert r.exit_code == 0, r.stderr
+            assert r.stdout.splitlines() == [
+                "On branch main",
+                "",
+                "Nothing to commit, working copy clean",
+            ]
+
+            new_head = repo.head.peel(pygit2.Commit).hex
+            assert new_head != orig_head
+
+            r = cli_runner.invoke(["checkout", "HEAD^"])
+
+            assert repo.head.peel(pygit2.Commit).hex == orig_head
+
+
+def test_approximated_types():
+    assert sqlserver_adapter.APPROXIMATED_TYPES == compute_approximated_types(
+        sqlserver_adapter.V2_TYPE_TO_MS_TYPE, sqlserver_adapter.MS_TYPE_TO_V2_TYPE
+    )
+
+
+def test_types_roundtrip(data_archive, cli_runner, new_sqlserver_db_schema):
```

Comment:
No, it just has a working copy with one of every sno datatype, including all different sized integers.
MSSQL is unique in that it's not able to roundtrip geometrytype or CRS (eg there's nowhere to store that a column is `POINT, 4326`) and so that diff has to be suppressed (eg, we take the "geometry" type from the working copy when generating a diff, but we simply copy the other info from the dataset, so that cannot differ and no change to it can be committed). If this wasn't working, any test that diffs geometry types would have a spurious meta diff, this one included.